test/async-observer.test.js

```diff
@@ -324,6 +324,86 @@ describe('async observer', function() {
       }
     );
   });
+
+  it('should call after operation hook on error', function(done) {
+    const context = {
+      req: {},
+    };
+    const operationError = new Error('The operation failed without result');
+    let passesBy = 0;
+
+    function fn(context, done) {
+      process.nextTick(() => {
+        done(operationError);
+      });
+    }
+
+    TestModel.observe('after execute error', function(ctx, next) {
+      passesBy++;
+      next();
+    });
+
+    TestModel.notifyObserversAround('execute', context, fn, (err, ctx) => {
+      passesBy.should.eql(1);
+      err.message.should.eql(operationError.message);
+      ctx.error.message.should.eql(operationError.message);
+      done();
+    });
+  });
+
+  it('should call after operation hook on error while overwriting error', function(done) {
```

Comment:
+1 for adding a test to verify what happens when both the operation and the hook report an error 👍